config/http_config.go

```diff
@@ -195,6 +199,12 @@ func (rt *bearerAuthRoundTripper) RoundTrip(req *http.Request) (*http.Response,
 	return rt.rt.RoundTrip(req)
 }
 
+func (rt *bearerAuthRoundTripper) CloseIdleConnections() {
+	if ci, ok := rt.rt.(closeIdler); ok {
```

Comment:
This is cleaner in Go 1.12 as it exposes this directly, without having to reach into the transport

Reply:
You refer to `client.CloseIdleConnections()`? I don't know how we can avoid going through the nested round-trippers until we call `Transport.CloseIdleConnections()`.

https://golang.org/src/net/http/client.go?s=27593:27632#L841

Reply:
Have we even switched over to go 1.12 everywhere? I feel like it might be a little early to use that. I think I'd prefer this approach at this point.

Reply:
Can we add a TODO at least to clean this up?

Reply:
I'm still not clear about what the TODO should be about... AFAICT consumers of the library using Go 1.12 will be able to do this and it will just work:

```go
client, err := config.NewClient(...)
...
client.CloseIdleConnections()
```

But again I don't see how we could avoid having `CloseIdleConnections()` for all the custom round-trippers if we want `client.CloseIdleConnections()` to be effective.